pkg/ccl/streamingccl/streamclient/partitioned_stream_client.go

```diff
@@ -80,15 +80,16 @@ var _ Client = &partitionedStreamClient{}
 
 // Create implements Client interface.
 func (p *partitionedStreamClient) Create(
-	ctx context.Context, tenantName roachpb.TenantName,
+	ctx context.Context, tenantName roachpb.TenantName, forSpanConfig bool,
 ) (streampb.ReplicationProducerSpec, error) {
 	ctx, sp := tracing.ChildSpan(ctx, "streamclient.Client.Create")
 	defer sp.Finish()
 
 	p.mu.Lock()
 	defer p.mu.Unlock()
 	var rawReplicationProducerSpec []byte
-	row := p.mu.srcConn.QueryRow(ctx, `SELECT crdb_internal.start_replication_stream($1)`, tenantName)
+	row := p.mu.srcConn.QueryRow(ctx, `SELECT crdb_internal.start_replication_stream($1,$2)`,
+		tenantName, forSpanConfig)
```

Comment:
is it allowed to use forSpanConfig=true with a tenantName other than system?

Reply:
good point. this query now fails if spanConfig==true and the tenanName is not system

Reply:
thanks, I thought so, in that case, what do you think about splitting the API to have 2 funcs, one for span configs and a "normal" one? so that you would not need the boolean. something like:
```
	Create(ctx context.Context, tenant roachpb.TenantName) (streampb.ReplicationProducerSpec, error)
	CreateForSpanConfigs(ctx context.Context) (streampb.ReplicationProducerSpec, error)
```
I don't think this is needed all over.. you'll decide where it makes more sense. overall I think it would be nicer than a bool, and callers will not be able to mix span configs and non-system tenant. another benefit might be that if we'll have `crdb_internal.start_replication_stream_for_span_configs` we'll get separate metrics for the 2 flavors (once we have metrics..).

Reply:
We chatted offline and decided that it's fine to keep the API as is-- there's only one caller of this interface, the stream ingestion job, and it will only ever call `Create()` with and without the bool. If we need to add another stream flavor, it would be nicer to add some sort of opts struct to `Create()` . @stevendanna when you get a chance, please do chime in here if you have opinions on how I'm changing the api.

Reply:
@lidorcarmel after thinking about this a bit more, I realized we need to persist the app tenant ID asap on the producer side. I think I need to modify this PR to do the following:
 - add a new field to the `SpanConfigsForTenantID` to the producer job record which persists the replicating app tenant id whose span config updates we want. The `TenantID` field will continue to be the system tenant id. 
 - modify the `startReplicationProducerJob()`, to persist the `SpanConfigsForTenantID` field if necessary. 
 
 From the client's perspective, we can either add:
 ```
 CreateForSpanConfigs(ctx context.Context, appTenant roachpb.TenantName) (streampb.ReplicationProducerSpec, error)
 ```
 or modify the current impl in the PR to:
 ```
	Create(ctx context.Context, appTenant roachpb.TenantName, forSpanConfigs bool) (streampb.ReplicationProducerSpec, error)
 ```
 i.e. the client always passes in the appTenant name, never a system tenant.
 
 We can discuss this on Monday.